File: Gemini/2. Estructura de Datos Pila (Stack)/Ruby/pila.rb
class Pila
  def initialize
    @items = []
  end

  def push(item)
    @items << item
  end

  def pop
    if !is_empty?
      @items.pop
    else
      "La pila está vacía"
    end
  end

  def is_empty?
    @items.empty?
  end
end

pila = Pila.new

loop do
  print "Ingrese una acción (push <elemento>, pop, verificar, salir): "
  accion = gets.chomp.split
  comando = accion[0].downcase

  case comando
  when "push"
    if accion.length > 1
      elemento = accion[1]
      pila.push(elemento)
      puts "Se agregó '#{elemento}' a la pila."
    else
      puts "Por favor, especifique el elemento a agregar."
    end
  when "pop"
    elemento = pila.pop
    puts "Se extrajo '#{elemento}' de la pila."
  when "verificar"
    if pila.is_empty?
      puts "La pila está vacía
      ."
    else
      puts "La pila no está vacía."
    end
  when "salir"
    break
  else
    puts "Comando no válido."
  end
en

File: Gemini/2. Estructura de Datos Pila (Stack)/Javascript/pila.js
class Pila {
  constructor() {
    this.items = [];
  }

  push(item) {
    this.items.push(item);
  }

  pop() {
    if (!this.isEmpty()) {
      return this.items.pop();
    } else {
      return "La pila está vacía";
    }
  }

  isEmpty() {
    return this.items.length === 0;
  }
}

const pila = new Pila();

function interactuarPila() {
  const accion = prompt("Ingrese una acción (push <elemento>, pop, verificar, salir):").split(" ");
  const comando = accion[0].toLowerCase();

  if (comando === "push") {
    if (accion.length > 1) {
      const elemento = accion[1];
      pila.push(elemento);
      console.log(`Se agregó '${elemento}' a la pila.`);
    } else {
      console.log("Por favor, especifique el elemento a agregar.");
    }
  } else if (comando === "pop") {
    const elemento = pila.pop();
    console.log(`Se extrajo '${elemento}' de la pila.`);
  } else if (comando === "verificar") {
    if (pila.isEmpty()) {
      console.log("La pila está vacía.");
    } else {
      console.log("La pila no está vacía.");
    }
  } else if (comando === "salir") {
    return;
  } else {
    console.log("Comando no válido.");
  }
  interactuarPila();
}

interactuarPila();